src/main/java/uk/sky/cirrus/locking/LockService.java

```diff
@@ -0,0 +1,77 @@
+package uk.sky.cirrus.locking;
+
+import com.datastax.driver.core.Row;
+import com.datastax.driver.core.Session;
+import com.datastax.driver.core.SimpleStatement;
+import com.datastax.driver.core.Statement;
+import com.datastax.driver.core.exceptions.DriverException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.sky.cirrus.locking.exception.CannotAcquireLockException;
+import uk.sky.cirrus.locking.exception.CannotReleaseLockException;
+
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class LockService {
+
+    private static final Logger log = LoggerFactory.getLogger(LockService.class);
+    private static final Statement SCHEMA_VERIFICATION_QUERY = new SimpleStatement("SELECT keyspace_name FROM system.schema_keyspaces WHERE keyspace_name = 'locks'");
+
+    static final AtomicBoolean SCHEMA_CREATED = new AtomicBoolean(false);
+
+    /**
+     * @param lockConfig {@code LockConfig} for configuring the lock to migrate the schema
+     * @param keyspace   Name of the keyspace which is to be used for migration
+     * @param session    Cassandra {@code Session} to use while acquiring the lock
+     * @return the {@code Lock} object
+     * @throws CannotAcquireLockException if instance cannot acquire lock within the specified time interval or execution of query to insert lock fails
+     */
+    public static Lock acquire(LockConfig lockConfig, String keyspace, Session session) {
+        while(!SCHEMA_CREATED.get()){
```

Comment:
This is overly complex for what gain? If you only read the schema created in a synchronized block you wouldn't need any atomic variables / busy while loops.